dspace-api/src/main/java/org/dspace/content/authority/LocalMetadataLookup.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.content.authority;
+
+import static org.dspace.content.authority.ChoiceAuthorityServiceImpl.CHOICES_INDEX_PREFIX;
+import static org.dspace.content.authority.ChoiceAuthorityServiceImpl.CHOICES_ORDER_PREFIX;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.FacetField;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.params.CommonParams;
+import org.apache.solr.common.params.FacetParams;
+import org.dspace.discovery.SearchUtils;
+import org.dspace.discovery.SolrSearchCore;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+
+public class LocalMetadataLookup implements ChoiceAuthority {
```

Comment:
We require JavaDocs for all public classes and methods.  This class needs to have a JavaDocs description to describe what it does and how it works

I also notice that there are no automated tests for this new class.  Would it be possible to provide some basic automated tests to prove this is working properly?  

(UPDATE: I do see now though that other "ChoiceAuthority" classes also don't have automated tests, so I'm not sure if that's just because they are old code, or if they are difficult to write tests for.)